rustyms-py/src/lib.rs

```diff
@@ -0,0 +1,1262 @@
+///
+#[pyclass]
+#[derive(Debug)]
+pub struct MolecularFormula(rustyms::MolecularFormula);
+
+#[pymethods]
+impl MolecularFormula {
+    // #[new]
+    // fn new(formula: &str) -> PyResult<Self> {
+    //     todo!()
+    // }
+
+    fn __repr__(&self) -> String {
+        format!("{}", self.0.to_string())
+    }
+
+    fn __str__(&self) -> String {
+        self.0.to_string()
+    }
+
+    /// Add the given element to this formula (while keeping it ordered and simplified)
+    ///
+    /// Parameters
+    /// ----------
+    /// element : Element
+    ///     The element to add.
+    /// isotope : int
+    ///     The isotope number of the element to add.
+    /// n : int
+    ///     The number of atoms of this element to add.
+    ///
+    fn add(&mut self, element: &Element, isotope: u16, n: i16) {
+        self.0.add((element.0.clone(), isotope, n));
+    }
+
+    /// Get the elements making this formula.
+    fn elements(&self) -> Vec<(Element, u16, i16)> {
+        self.0
+            .elements()
+            .iter()
+            .map(|(e, i, n)| (Element(e.clone()), *i, *n))
+            .collect()
+    }
+
+    // TODO: Get this working (issues with slice type and pyo3)
+    // /// Create a new molecular formula with the given global isotope modifications.
+    // fn with_global_isotope_modifications(&self, substitutions: Vec<(Element, u16)>) -> Self {
+    //     let substitutions = substitutions
+    //         .iter()
+    //         .map(|(e, i)| (e.0.clone(), *i))
+    //         .collect::<Vec<_>>()
+    //         .as_slice();
+    //     MolecularFormula(self.0.with_global_isotope_modifications(&substitutions))
+    // }
+
+    /// Get the number of electrons (the only charged species, any ionic species is saved as that element +/- the correct number of electrons). The inverse of that number is given as the charge.
+    ///
+    /// Returns
+    /// -------
+    /// int
+    ///
+    fn charge(&self) -> i16 {
+        self.0.charge()
+    }
+
+    /// The mass of the molecular formula of this element, if all element species (isotopes) exists
+    ///
+    /// Returns
+    /// -------
+    /// float | None
+    ///
+    fn monoisotopic_mass(&self) -> Option<f64> {
+        match self.0.monoisotopic_mass() {
+            Some(mass) => Some(mass.value),
+            None => None,
+        }
+    }
+
+    /// The average weight of the molecular formula of this element, if all element species (isotopes) exists.
+    ///
+    /// Returns
+    /// -------
+    /// float | None
+    ///
+    fn average_weight(&self) -> Option<f64> {
+        match self.0.average_weight() {
+            Some(mass) => Some(mass.value),
+            None => None,
+        }
+    }
+
+    /// The most abundant mass, meaning the isotope that will have the highest intensity.
+    ///
+    /// Returns
+    /// -------
+    /// float | None
+    ///
+    fn most_abundant_mass(&self) -> Option<f64> {
```

Comment:
This one is broken upstream as well, so maybe remove this from the public api for now (I should maybe to the same on the rust side)